src/Type/EntityStorage/EntityStorageDynamicReturnTypeExtension.php

```diff
@@ -58,39 +61,24 @@ public function getTypeFromMethodCall(
         Scope $scope
     ): \PHPStan\Type\Type {
         $callerType = $scope->getType($methodCall->var);
-
-        if (!$callerType instanceof EntityStorageType) {
-            return ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
-        }
-
-        $type = $this->entityDataRepository->get($callerType->getEntityTypeId())->getClassType();
-        if ($type === null) {
+        if (!$callerType instanceof ObjectType) {
             return ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
         }
-        if (\in_array($methodReflection->getName(), ['load', 'loadUnchanged'], true)) {
-            return TypeCombinator::addNull($type);
-        }
-
-        if (\in_array($methodReflection->getName(), ['loadMultiple', 'loadByProperties'], true)) {
-            if ($callerType instanceof ConfigEntityStorageType) {
-                return new ArrayType(new StringType(), $type);
-            }
 
-            return new ArrayType(new IntegerType(), $type);
-        }
         if ($methodReflection->getName() === 'getQuery') {
             $returnType = ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
             if (!$returnType instanceof ObjectType) {
                 return $returnType;
             }
-            if ($callerType instanceof ContentEntityStorageType) {
+
+            if ((new ObjectType(ContentEntityStorageInterface::class))->isSuperTypeOf($callerType)->yes()) {
```

Comment:
This is impossible to grok from the diff. All I did was move `getQuery` above the check for the entity type class. Because we don't really care about the entity type class if we have an object type that's a storage class. Then we can determine the kind of query will be executed, which does not require knowing the entity type class.

Reply:
I'm thinking we should do this in its own PR. So make `EntityStorageGetQueryDynamicReturnTypeExtension` to just handle this `getQuery` method since we do not require the entity type class. Then the `EntityStorageDynamicReturnTypeExtension` only handles methods which return a single instance or array of the entity type class.